bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```diff
@@ -0,0 +1,294 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.qbus.internal.protocol.QbusCommunication;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link QbusBridgeHandler} is the handler for a Qbus controller
+ *
+ * @author Koen Schockaert - Initial Contribution
+ */
+
+@NonNullByDefault
+public class QbusBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(QbusBridgeHandler.class);
+
+    private @Nullable QbusCommunication qbusComm;
+
+    protected @Nullable QbusConfiguration config;
+
+    private @Nullable ScheduledFuture<?> refreshTimer;
+
+    public QbusBridgeHandler(Bridge Bridge) {
+        super(Bridge);
+    }
+
+    /**
+     * Initialize the bridge
+     */
+    @Override
+    public void initialize() {
+        readConfig();
+
+        InetAddress addr;
+        Integer port = getPort();
+        Integer refresh = getRefresh();
+
+        if (port == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "No port defined for Qbus Server");
+            return;
+        }
+
+        try {
+            addr = InetAddress.getByName(getAddress());
+            if (addr == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        "No ip address defined for Qbus Server");
+                return;
+            } else {
+                createCommunicationObject(addr, port);
+            }
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Incorrect ip address set for Qbus Server");
+        }
+
+        if (refresh != null) {
+            this.setupRefreshTimer(refresh);
+        }
+    }
+
+    /**
+     * Sets the Bridge call back
+     */
+    private void setBridgeCallBack() {
+        if (qbusComm != null) {
+            qbusComm.setBridgeCallBack(this);
+        }
+    }
+
+    /**
+     * Create communication object to Qbus server and start communication.
+     *
+     * @param addr : IP address of Qbus server
+     * @param port : Communication port of QbusServer
+     */
+    private void createCommunicationObject(InetAddress addr, int port) {
+        scheduler.submit(() -> {
+
+            qbusComm = new QbusCommunication();
+
+            setBridgeCallBack();
+
+            if (qbusComm != null) {
+                try {
+                    qbusComm.startCommunication();
+                } catch (InterruptedException | IOException e) {
+                    logger.warn("Error on restaring communication.");
```

Comment:
It's good practice to append the exception's message to the log message. Should the Thing set to OFFLINE here?
```suggestion
                    logger.warn("Error on restaring communication: {}", e.getMessage());
```